src/core/control/jobs/RenderJob.cpp

```diff
@@ -79,12 +81,10 @@ void RenderJob::run() {
 
         std::lock_guard lock(this->view->drawingMutex);
         std::swap(this->view->crBuffer, newBuffer);
+        this->view->repaintPage();
```

Comment:
I'm not 100% sure, but this may need to be wrapped in `Util::execInUiThread(...)`. The `RenderJob` runs in a separate thread from the GTK thread, and GTK is not thread safe.

Reply:
How are you calling `Util::execInUiThread`?